packages/react/focus-scope/src/FocusScope.tsx

```diff
@@ -53,6 +53,35 @@ function FocusScope(props: FocusScopeProps) {
     },
   }).current;
 
+  // Takes care of containing focus if focus is moved outside programmatically for example
```

Comment:
I moved this effect first so that the `handleFocusIn` can catch the initial focus from `onMountAutoFocus` and thus record the initial mount focus as `lastFocusedElementRef.current`.